atef/config.py

```diff
@@ -112,34 +127,46 @@ class ConfigurationFile:
     """
     A configuration file comprised of a number of devices/PV configurations.
     """
+    #: atef configuration file version information.
+    version: Literal[0] = field(default=0, metadata=apischema.metadata.required)
```

Comment:
How is this going to be used? Increment this number every time we update the schema? Use old schema info if we read a file with an old number?

Depending on what we want to do it might be appropriate to automate this via tagging

Reply:
If breaking changes are made in the schema, we bump the version.
Then we make a shim to convert older schemas to newer ones.
It may be possible to automate, but I think it might be more trouble than it's worth.

Reply:
`breaking changes` was the keyword I was missing that makes this useful/sustainable to update